lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Transactions.hs

```diff
@@ -533,6 +551,190 @@ spec = do
+        -- 4. produce delegation address (here second address of index 1 and for network tag 1):
+        --- root.prv \
+        --- | cardano-address key child 1852H/1815H/0H/0/1 \
+        --- | cardano-address key public \
+        --- | cardano-address address payment --network-tag 1 \
+        --- | cardano-address address delegation $(cat stake.prv | cardano-address key public)
+        --- --> addr1q9wteuqmnkywcz9zefjyp0wn3tctz9xgxm8fpcmyzn9uypknudck0fzve4346yytz3wpwv9yhlxt7jwuc7ytwx2vfkyq75lt2a
+        -- 5. produce corresponding private key (used for witnesses):
+        --- cat root.prv | cardano-address key child 1852H/1815H/0H/0/1 --base16
+        -- --> 400cd8a0c260e5aafe1768bf077815351fedf1bb698e1fb4a32903d52f006541c9dfe5b6c62f5c30aa958711233696d2d3f8486547a9e2f928d064f5752acfa81d028f8516adad09a8efe3a0209e7e75d6bc61265f48d7471acdd2bf7ee7b23d
+        -- 6. produce change address:
+        --- root.prv \
+        --- | cardano-address key child 1852H/1815H/0H/1/0 \
+        --- | cardano-address key public \
+        --- | cardano-address address payment --network-tag 1 \
+        --- | cardano-address address delegation $(cat stake.prv | cardano-address key public)
+        -- --> addr1q8zamr0gglxc7dsnhdwfp0pg9hac50zc9wrgh5gmwxm5f7xnudck0fzve4346yytz3wpwv9yhlxt7jwuc7ytwx2vfkyquaaj4r
+        -- (d) mnemonic15 wallet received during the first transaction 10 ada to address
+        -- and here is its witness (details of getting it above)
+        let wit =
+                "400cd8a0c260e5aafe1768bf077815351fedf1bb698e1fb4a32903d52f0065\
+                \41c9dfe5b6c62f5c30aa958711233696d2d3f8486547a9e2f928d064f5752a\
+                \cfa81d028f8516adad09a8efe3a0209e7e75d6bc61265f48d7471acdd2bf7e\
+                \e7b23d"
+        let (Right blob) =
+                constructTxFromCardanoTransactions (fromIntegral feeMin1) txid txix out1 addr1 amt1 addr2 wit
+        let baseOk = Base64
+        let encodedSignedTx = T.decodeUtf8 $ convertToBase baseOk blob
+        let payloadExt = NonJson . BL.fromStrict . toRawBytes baseOk
+        let headers = Headers [ ("Content-Type", "application/octet-stream") ]
+        r <- request
+            @ApiTxId ctx Link.postExternalTransaction headers (payloadExt encodedSignedTx)
+        verify r
+            [ expectSuccess
+            , expectResponseCode HTTP.status202
+            ]
+
+        ------ beginning of conventional submitting of tx - to be removed in later commits
+{--
+
+        r6 <- request @(ApiTransaction n) ctx
+            (Link.createTransaction @'Shelley wSrc) Default payload2
+
+        verify r6
+            [ expectSuccess
+            , expectResponseCode HTTP.status202
+            , expectField (#amount . #getQuantity) $
+                between (feeMin1 + amt1, feeMax1 + amt1)
+            , expectField (#direction . #getApiT) (`shouldBe` Outgoing)
+            , expectField (#status . #getApiT) (`shouldBe` Pending)
+            ]
+--}
+{--
```

Comment:
this should be uncommented and should pass. now I have : 
```
      uncaught exception: IOException of type UserError
       user error (expected a successful response but got an error: DecodeFailure "Error in $: parsing Cardano.Wallet.Api.Types.ApiWallet(ApiWallet) failed, key \"id\" not found: Response {responseStatus = Status {statusCode = 409, statusMessage = \"Conflict\"}, responseVersion = HTTP/1.1, responseHeaders = [(\"Transfer-Encoding\",\"chunked\"),(\"Date\",\"Sat, 15 Aug 2020 18:40:10 GMT\"),(\"Server\",\"Warp/3.3.5\")], responseBody = \"{\\\"code\\\":\\\"wallet_already_exists\\\",\\\"message\\\":\\\"This operation would yield a wallet with the following id: b062e8ccf3685549b6c489a4e94966bc4695b75b However, I already know of a wallet with this id.\\\"}\", responseCookieJar = CJ {expose = []}, responseClose' = ResponseClose}")
```
strange, but maybe something more needs to be changed to accommodate external tx

Reply:
this is from line 717

Reply:
now it works